### ISSUE TITLE
[BUG]: Unrecognized option -m when attempting to run matmul.mojo in the examples folder

### ISSUE THREAD

**Issue by gsnyder2007** [modularml/mojo#1079]
### Bug description

When attempting to run "mojo matmul.mojo" in the examples folder I receive the following error,

(venv) *[main][~/Projects/mojo/examples]$ mojo matmul.mojo                   rbenv:system 
Numpy not found, installing...
/Users/gsnyder/.modular/pkg/packages.modular.com_mojo/bin/mojo: error: unrecognized argument '-m'
Unhandled exception caught during execution: Command '['/Users/gsnyder/.modular/pkg/packages.modular.com_mojo/bin/mojo', '-m', 'pip', 'install', 'numpy']' returned non-zero exit status 1.
mojo: error: execution exited with a non-zero result: 1

Here are more details regarding what I did and my attempts to debug the issue:

After installing mojo ran into an issue when attempting to run matmul.mojo,
1. Set my mojo/python venv using: source ~/.modular/pkg/packages.modular.com_mojo/venv/bin/activate
2. Did a "git pull" to update mojo Git repo
3. Then tried to run matmul as per https://www.modular.com/blog/mojo-is-now-available-on-mac?utm_medium=email&_hsmi=279018192&_hsenc=p2ANqtz--Kcpn2dkYgxVkzJlGy5nTUrH2iYM9GGQujvXyKvtXWuXXs-O3dYzTR8tCCpzQcFhkkyHe8AN31fAQQVU1A5qv6grsEviGfjNej5VEcZXBerq_O4f0&utm_content=279018192&utm_source=hs_email

---
(venv) *[main][~/Projects/mojo/examples]$ mojo matmul.mojo                   rbenv:system 
Numpy not found, installing...
/Users/gsnyder/.modular/pkg/packages.modular.com_mojo/bin/mojo: error: unrecognized argument '-m'
Unhandled exception caught during execution: Command '['/Users/gsnyder/.modular/pkg/packages.modular.com_mojo/bin/mojo', '-m', 'pip', 'install', 'numpy']' returned non-zero exit status 1.
mojo: error: execution exited with a non-zero result: 1
---

Hmm, why is mojo trying to install numpy? Looking at the mojo --help I don't see any "-m" option. Does this version support a -m option?

(venv) *[main][~/Projects/mojo/examples]$ mojo --version                     rbenv:system 
mojo 0.4.0 (9e33b013)

So, I decided to do a test and install numpy manually using pip,

(venv) *[main][~/Projects/mojo/examples]$ pip install numpy

....and it failed again (same error as above)

I checked and didn't see any reference to numpy in matmul.mojo. Maybe something has a transitive dependency on numpy? so mojo tries to install it?

Then I decided to try and build matmul.mojo manually,

(venv) *[main][~/Projects/mojo/examples]$ mojo build matmul.mojo             rbenv:system 

That worked. Let's run it,

(venv) *[main][~/Projects/mojo/examples]$ ./matmul                           rbenv:system 
Mojo/Python interoperability error: Unable to locate a suitable libpython, please set `MOJO_PYTHON_LIBRARY`
[1]    52051 segmentation fault  ./matmul

Oops. Ok, so I don't have MOJO_PYTHON_LIBRARY set.

(venv) *[main][~/Projects/mojo/examples]$ which python                       rbenv:system 
/Users/gsnyder/.modular/pkg/packages.modular.com_mojo/venv/bin/python
(venv) *[main][~/Projects/mojo/examples]$ export MOJO_PYTHON_LIBRARY=/Users/gsnyder/.modular/pkg/packages.modular.com_mojo/venv/bin/python
(venv) *[main][~/Projects/mojo/examples]$ ./matmul                           rbenv:system 
Numpy not found, installing...
Numpy not found, installing...
Numpy not found, installing...
Numpy not found, installing...
.
.
.

And so we are back to where we started. Something really wants numpy and has decided to install it, even though it is installed into the venv (from python's perspective, but not mojo's?)

(venv) *[main][~/Projects/mojo/examples]$ mojo -m pip install mojo           rbenv:system 
mojo: error: unrecognized argument '-m'

I'm on a MacBook Pro 14 inch, 2023, M2 Max, Ventura 13.4. 

### Steps to reproduce

Was following steps in this article announcing support for Apple Silicon,

https://www.modular.com/blog/mojo-is-now-available-on-mac

...and the code I was attempting to run is here,

https://github.com/modularml/mojo/blob/main/examples/matmul.mojo

### System information

```shell
- What OS did you do install Mojo on ?
- Provide version information for Mojo by pasting the output of `mojo -v`
- Provide Modular CLI version by pasting the output of `modular -v`

(venv) *[main][~/Projects/mojo/examples]$ mojo --version                     rbenv:system 
mojo 0.4.0 (9e33b013)
(venv) *[main][~/Projects/mojo/examples]$ modular -v                         rbenv:system 
modular 0.2.1 (5144fffe)

Apple M2 Max running Ventura 13.4
```

**Comment by bzcheeseman** [comment]
It looks like you have something trying to install `numpy`:
```
Command '['/Users/gsnyder/.modular/pkg/packages.modular.com_mojo/bin/mojo', '-m', 'pip', 'install', 'numpy']' 
```

Thing is, you can't use `mojo` to do a `pip install` :)

My suspicion is your `rbenv` thing - not sure what that is but that seems like it's trying to handle dependencies automatically for you?

**Comment by gsnyder2007** [comment]
I may have figured this out and I think it's an issue in the example code combined with my environment, i.e. not an issue with mojo. 

The rbenv stuff is from https://ohmyz.sh/ BTW and I don't think it's causing the issue (still investigating). And, I did try installing numpy manually (into the modular venv) but to no avail. So, back to my theory which is the example code and my environment are conspiring to create the issue.

In pymatmul.py there are these lines which look for numpy and if not present try to install it,

if not find_spec("numpy"):
    print("Numpy not found, installing...")
    subprocess.check_call([sys.executable, "-m", "pip", "install", "numpy"])

The reference to sys.executable ends up resolving to mojo (in my environment). I confirmed this by changing these lines to,

if not find_spec("numpy"):
    print("Numpy not found, installing...")
    subprocess.check_call(["python", "-m", "pip", "install", "numpy"])

 Here's what happened when I then tried to run matmul.mojo, with sys.executable replaced with "python",

(venv) *[main][~/Projects/mojo/examples]$ mojo matmul.mojo
Numpy not found, installing...
Looking in indexes: https://pypi.python.org/simple/, https://1su0wV-Wzx3hl8J69MoVyh6oMmftR8Kkj4:****@pypi.fury.io/gsnyder-ootw/
Requirement already satisfied: numpy in /Users/gsnyder/.modular/pkg/packages.modular.com_mojo/venv/lib/python3.11/site-packages (1.26.1)

[notice] A new release of pip is available: 23.1.2 -> 23.3
[notice] To update, run: pip install --upgrade pip
Unhandled exception caught during execution: No module named 'numpy'
mojo: error: execution exited with a non-zero result: 1

So, this got further than before but still failed when numpy wasn't found (later). I'm going to assume that's also an environmental error where my environment isn't seeing the numpy that was installed. 

I'm going to further investigate and would ask we pause this issue (from your POV) for now as it may not be a good use of your time. Or close it and I can reopen something later if/when I confirm what the environment issue really is. I think at this point it's clear this is NOT an issue with the mojo executable itself. We can always reopen later if I'm wrong about that. :-)
